site/src/pages/docs/advanced-scala.mdx

```diff
@@ -0,0 +1,161 @@
+import com.linecorp.armeria.server.Server
+
+import java.util.concurrent.CompletableFuture
+import scala.concurrent.Future
+
+val server =
+  Server
+    .builder()
+    .service("/", (ctx, req) => HttpResponse.of(200))
+    .build()
+
+val javaFuture: CompletableFuture[Void] = server.start()
+val scalaFuture: Future[Unit] = javaFuture.toScala // 👈
+```
+
+You can also convert a Scala `Future` to a Java `CompletionStage` using the `toJava` method:
+
+```scala
+import java.util.concurrent.CompletionStage
+val javaFuture: CompletionStage[Void] = scalaFuture.toJava
+```
+
+You'll also find the extension method `toHttpResponse` added to `Future[HttpResponse]` and
+`CompletionStage[HttpResponse]` when converting an asynchronous result into an <type://HttpResponse>:
+
+```scala
+import com.linecorp.armeria.common.MediaType
+import com.linecorp.armeria.scala.implicits._
+
+val future: Future[String] = ...
+val response =
+  future
+    .map { value => HttpResponse.of(MediaType.PLAIN_TEXT_UTF_8, value) }
+    .toHttpResponse // 👈
+```
+
+<Tip>
+
+You might wonder why we implemented our own extension methods rather then letting you use
+`FutureConverters` in [`scala-java8-compat`](https://github.com/scala/scala-java8-compat).
+It is because `armeria-scala` provides richer and more natural conversions, such as:
+
+- Conversion to <type://HttpResponse>
+- Conversion from `CompletionStage[Void]` to `Future[Unit]`
+- Conversion from `Future[Unit]` to `CompletionStage[Void]`
+
+</Tip>
+
+## `directExecutionContext`
+
+In an asynchronous system that the entire application logic runs on event loops, it is often useful
+in term of performance to invoke the callbacks attached to a `Future` directly rather than submitting
+the callbacks to another `ExecutionContext`. You can use `directExecutionContext` in such a case:
+
+```scala
+import com.linecorp.armeria.common.HttpRequest
+import com.linecorp.armeria.scala.directExecutionContext
+import com.linecorp.armeria.scala.implicits._
+import com.linecorp.armeria.server.HttpService
+import com.linecorp.armeria.server.ServiceRequestContext
+
+
+class MyHttpService extends HttpService {
+  override def serve(ctx: ServiceRequestContext, req: HttpRequest): HttpResponse = {
+    // Perform some asynchronous operation.
+    val future: Future[String] = ...
+
+    // Convert the result to a response.
+    future
+      .map { value =>
+        HttpResponse.of(MediaType.PLAIN_TEXT_UTF_8, value)
+      }(directExecutionContext) // 👈
+      .toHttpResponse
+  }
+}
+```
+
+## Context-aware `ExecutionContext`
+
+You can use the `eventLoopExecutionContext` extension method in <type://RequestContext> to specify a Scala
+`ExecutionContext` that submits all tasks to the current context's event loop thread:
+
+```scala
+import com.linecorp.armeria.scala.implicits._
+
+val ctx = ServiceRequestContext.current
+Future {
+  // Do some non-blocking job here.
+}(ctx.eventLoopExecutionContext) // 👈
```

Comment:
How about adding an implicit version too?
```scala
// Or 
implicit val ec: ExecutionContext = ctx.eventLoopExecutionContext

Future {
  // Do some non-blocking job here.
}
```

Reply:
Wouldn't that make every `map/flatmap()` scheduled via the execution context? I'm not sure if it's a good practice in our case, so I only used an implicit EC for `directExecutionContext`.

Reply:
Just changed as you suggested. :smile: